tests/framework/common.go

```diff
@@ -442,6 +442,7 @@ func setMeshConfigToDefault(instOpts InstallOSMOpts, meshConfig *configv1alpha2.
 	meshConfig.Spec.Sidecar.LogLevel = instOpts.EnvoyLogLevel
 	meshConfig.Spec.Sidecar.MaxDataPlaneConnections = 0
 	meshConfig.Spec.Sidecar.ConfigResyncInterval = "0s"
+	meshConfig.Spec.Sidecar.LocalProxyMode = instOpts.LocalProxyMode
```

Comment:
@keithmattix isn't this set as the default? Wondering why this needs to be explicitly set without looking further at the test code.

Reply:
LocalProxyMode is changed for the "Test traffic flowing from client to a server with a podIP bind" part of the test to v1alpha2.LocalProxyModePodIP, the previous 2 ginkgo.contexts use the default Proxy Mode. 

The way the test was before, the new value for LocalProxyMode was only being set for fresh installs. For the change to also be made for NoInstall, the value needs to be explicitly changed in setMeshConfigToDefault.

Reply:
Because we don't have a helper function to access the LocalProxyMode field, there's no default being set outside of a Helm install.

Reply:
@keithmattix Why don't we default its value in the CRD schema?

Reply:
[We do](https://github.com/openservicemesh/osm/blob/main/cmd/osm-bootstrap/crds/config_meshconfig.yaml#L125). But unless we're using an openapi unmarshaler, those defaults won't take effect

Reply:
We could add a default in the json unmarshaler, but that also means we would need to change the API version if we change those defaults. This seems more like a test issue where the control plane is reused between tests.

Reply:
Yep, that's one of the downsides to adding not bumping a new version. This will also happen when users bump their OSM version, but because LocalProxyConfig is an opt-in behavior, we're fine with the 3rd case (unset) being the default.